pkg/appmanager/resourceConfig.go

```diff
@@ -220,6 +220,31 @@ func formatIngressPoolName(namespace, ingName, svc string) string {
 	return fmt.Sprintf("ingress_%s_%s_%s", namespace, ingName, svc)
 }
 
+func getRouteCanonicalService(route *routeapi.Route) string {
+	return route.Spec.To.Name
+}
+
+// return the services associated with a route
+func getRouteServiceNames(route *routeapi.Route) []string {
+	numOfSvcs := 1
+	if route.Spec.AlternateBackends != nil {
+		numOfSvcs += len(route.Spec.AlternateBackends)
+	}
+
+	svcs := make([]string, numOfSvcs)
+
+	svcIdx := 0
+	if route.Spec.AlternateBackends != nil {
+		for i, svc := range route.Spec.AlternateBackends {
+			svcIdx = i
+			svcs[svcIdx] = svc.Name
+		}
+	}
+	svcs[svcIdx] = getRouteCanonicalService(route)
```

Comment:
Won't this overwrite the last alternate backend in the list, since svcIdx isn't incremented once the loop has exited?

Reply:
I believe it would.  I moved it from being the first to the last, so I forgot this important detail.  Good catch.

Reply:
I just noticed this is an older version of the code.  I forgot to push the latest.  Will have a new version pushed in a minute or two.